From f97b080dcd0b8b36a170604d0a772f0f6d7fcb08 Mon Sep 17 00:00:00 2001
From: Alexander Stein <alexander.stein@tq-group.com>
Date: Wed, 2 Nov 2022 14:37:21 +0100
Subject: [PATCH] Fix GLES3 detection

Checking pkg-cfg version does not work on NXP or TI graphics stack. Check
for actual GLES3/gl3.h instead.

Upstream-Issue: https://gitlab.freedesktop.org/mesa/kmscube/-/issues/8
---
 common.h    | 8 ++++++++
 meson.build | 9 ++++-----
 2 files changed, 12 insertions(+), 5 deletions(-)

diff --git a/common.h b/common.h
index c5d624a..a34801f 100644
--- a/common.h
+++ b/common.h
@@ -207,11 +207,19 @@ init_cube_video(const struct gbm *gbm, const char *video, int samples)
 }
 #endif
 
+#if defined(HAVE_GLES3)
 void init_perfcntrs(const struct egl *egl, const char *perfcntrs);
 void start_perfcntrs(void);
 void end_perfcntrs(void);
 void finish_perfcntrs(void);
 void dump_perfcntrs(unsigned nframes, uint64_t elapsed_time_ns);
+#else
+static void init_perfcntrs(const struct egl *egl, const char *perfcntrs) {}
+static void start_perfcntrs(void) {}
+static void end_perfcntrs(void) {}
+static void finish_perfcntrs(void) {}
+static void dump_perfcntrs(unsigned nframes, uint64_t elapsed_time_ns) {}
+#endif
 
 #define NSEC_PER_SEC (INT64_C(1000) * USEC_PER_SEC)
 #define USEC_PER_SEC (INT64_C(1000) * MSEC_PER_SEC)
diff --git a/meson.build b/meson.build
index 518006f..608f2f9 100644
--- a/meson.build
+++ b/meson.build
@@ -44,7 +44,6 @@ sources = files(
   'frame-512x512-NV12.c',
   'frame-512x512-RGBA.c',
   'kmscube.c',
-  'perfcntrs.c',
 )
 
 cc = meson.get_compiler('c')
@@ -54,10 +53,10 @@ dep_libdrm = dependency('libdrm', version : '>=2.4.71')
 dep_gbm = dependency('gbm', version : '>=13.0')
 dep_egl = dependency('egl')
 dep_gles2 = dependency('glesv2')
-dep_gles3 = dependency('glesv2', version : '>= 3', required : false)
+gles3_h = cc.has_header('GLES3/gl3.h', dependencies: dep_gles2)
 
-if dep_gles3.found()
-  sources += files('cube-shadertoy.c')
+if gles3_h
+  sources += files('cube-shadertoy.c', 'perfcntrs.c')
   add_project_arguments('-DHAVE_GLES3', language : 'c')
   message('GLES3 supported; shadertoy & texturator are included in this build')
 else
@@ -103,7 +102,7 @@ endif
 
 executable('kmscube', sources, dependencies : dep_common, install : true)
 
-if dep_gles3.found()
+if gles3_h
   executable('texturator', files(
     'common.c',
     'drm-legacy.c',
